File: PokerGameTracker/src/PokerGameTracker.Core/Model/SessionAggregate/Session.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace PokerGameTracker.Core;

public class Session
{
    public Guid SessionId { get; set; }
    public Guid UserId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public GameType GameType { get; set; }
    public DateTime StartTime { get; set; } = DateTime.UtcNow;
    public DateTime? EndTime { get; set; }
    public decimal BuyIn { get; set; }
    public decimal? CashOut { get; set; }
    public string? Location { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<Hand> Hands { get; set; } = new List<Hand>();
}

File: ConversationStarterApp/src/ConversationStarterApp.Web/projects/api/src/lib/models/session.ts
export interface Session {
  sessionId: string;
  userId: string;
  title: string;
  startTime: Date;
  endTime?: Date;
  participants?: string;
  promptsUsed?: string;
  notes?: string;
  wasSuccessful: boolean;
  createdAt: Date;
  updatedAt?: Date;
  duration?: string;
}

export interface CreateSessionRequest {
  userId: string;
  title: string;
  startTime?: Date;
  participants?: string;
  promptsUsed?: string;
}

export interface UpdateSessionRequest {
  sessionId: string;
  title: string;
  participants?: string;
  promptsUsed?: string;
  notes?: string;
  wasSuccessful: boolean;
}
